stl/inc/xstring

```diff
@@ -3980,6 +3980,21 @@ public:
         }
     }
 
+#if _HAS_CXX23
+    template <class _Operation>
+    constexpr void resize_and_overwrite(_CRT_GUARDOVERFLOW const size_type _Newsize, _Operation _Op) {
+        if (_Mypair._Myval2._Myres < _Newsize) {
+            _Reallocate_grow_by(
+                _Newsize - size(),
+                [](_Elem* const _New_ptr, const _Elem* const _Old_ptr, const size_type _Oldsize,
+                    const size_type _Newsize) { _Traits::copy(_New_ptr, _Old_ptr, (_STD min) (_Oldsize, _Newsize)); },
+                _Newsize);
+        }
+
+        _Eos(_STD move(_Op)(data(), _Newsize));

```

Comment:
I think the returned value should be checked with `_STL_VERIFY` in case of `_CONTAINER_DEBUG_LEVEL > 0`

Reply:
I believe that is an actual bug. Not only can we overflow and should chck against that, but we could also overflow `size_type`.
```suggestion
       const auto _Provided_size = _Convert_size<size_type>(_STD move(_Op)(data(), _Newsize));
       _STL_VERIFY(_Provided_size <= _Newsize);
        _Eos(_Provided_size);
```

We should also add some death tests to tests\std\tests\VSO_0830211_container_debugging_range_checks\test.cpp

Reply:
We can't overflow `size_type` because the result can't be greater than `_Newsize`.

Reply:
It is UB if function lies. I think we only should catch it in `_CONTAINER_DEBUG_LEVEL > 0`